openeogeotrellis/load_stac.py

```diff
@@ -505,6 +506,25 @@ def intersects_temporally(interval) -> bool:
     return GeopysparkDataCube(pyramid=gps.Pyramid(levels), metadata=metadata)
 
 
+def get_best_url(asset):
+    """
+    Relevant doc: https://github.com/stac-extensions/alternate-assets
+    """
+    alternate = asset.extra_fields.get("alternate")
+    if alternate:
+        for key, alternate_local in alternate.items():
+            if key != "local":
+                continue
+            href = alternate_local.get("href")
+            # Checking if file exists takes around 10ms on /data/MTDA mounted on laptop
+            # Checking if URL exists takes around 100ms on https://services.terrascope.be
+            # Checking if URL exists depends also on what Datasource is used in the scala code.
+            # That would be hacky to predict here.
+            if Path(urlparse(href).path).exists():
```

Comment:
This looks pretty weird: you just check if the path of a href exists locally.
Can you give an example where that would make sense?

Reply:
It only checks when for a "local" alternate url. Example data:
```json
{
    "B02_10m": {
        "href": "https://services.terrascope.be/download/CGS_S2_L2A/2024/07/18/S2A_MSIL2A_20240718T123311_N0510_R052_T28VDR_20240718T165647/S2A_MSIL2A_20240718T123311_N0510_R052_T28VDR_20240718T165647.SAFE/GRANULE/L2A_T28VDR_A047382_20240718T123307/IMG_DATA/R10m/T28VDR_20240718T123311_B02_10m.jp2",
        "type": "image/jp2",
        // ...
        "alternate": {
            "local": {
                "href": "file:///data/MTDA/CGS_S2/CGS_S2_L2A/2024/07/18/S2A_MSIL2A_20240718T123311_N0510_R052_T28VDR_20240718T165647/S2A_MSIL2A_20240718T123311_N0510_R052_T28VDR_20240718T165647.SAFE/GRANULE/L2A_T28VDR_A047382_20240718T123307/IMG_DATA/R10m/T28VDR_20240718T123311_B02_10m.jp2"
            }
        },
        // ...
    }
}
```
https://stac.terrascope.be/collections/sentinel-2-l2a/items/S2A_MSIL2A_20240718T123311_N0510_R052_T28VDR_20240718T165647

Reply:
I would then also include a check that the protocol is `file://`

Reply:
Added. 
Btw, not sure if it is cost effective to add an auto test for this. Breaking `def load_stac` in pieces will be a big effort, and adding a semi-external stac catalog to integration tests might break in the future.

Reply:
you can just unit-test  `get_best_url` directly, right? Should be pretty simple (no need for complex mocking), and acts as added context/docs what this feature is about.

Reply:
Creating the stac asset directly did not initialize the alternate properties. Added a test now that loads a small region of a mocked catalog.